src/tests/performance/connections.rs

```diff
@@ -0,0 +1,218 @@
+    // start node
+    let mut node = Node::builder()
+        .max_peers(MAX_PEERS as usize)
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .unwrap();
+    let node_addr = node.addr();
+
+    for synth_count in synth_counts {
+        // setup metrics recorder
+        let test_metrics = TestMetrics::default();
+        // register metrics
+        metrics::register_counter!(METRIC_ACCEPTED);
+        metrics::register_counter!(METRIC_TERMINATED);
+        metrics::register_counter!(METRIC_REJECTED);
+        metrics::register_counter!(METRIC_ERROR);
+
+        let mut synth_handles = Vec::with_capacity(synth_count as usize);
+        let mut synth_exits = Vec::with_capacity(synth_count as usize);
+        let (handshake_tx, mut handshake_rx) =
+            tokio::sync::mpsc::channel::<()>(synth_count as usize);
+
+        let test_start = Instant::now();
+
+        // start synthetic nodes
+        for _ in 0..synth_count {
+            let (exit_tx, exit_rx) = tokio::sync::oneshot::channel::<()>();
+            synth_exits.push(exit_tx);
+
+            let synth_handshaken = handshake_tx.clone();
+            // Synthetic node runs until it completes or is instructed to exit
+            synth_handles.push(tokio::spawn(async move {
+                tokio::select! {
+                    _ = exit_rx => {},
+                    _ = simulate_peer(node_addr, synth_handshaken) => {},
+                };
+            }));
+        }
+
+        // Wait for all peers to indicate that they've completed the handshake portion
+        // or the iteration timeout is exceeded.
+        let _ = tokio::time::timeout(MAX_ITER_TIME, async move {
+            for _ in 0..synth_count {
+                handshake_rx.recv().await.unwrap();
+            }
+        })
+        .await;
+
+        // Send stop signal to peer nodes. We ignore the possible error
+        // result as this will occur with peers that have already exited.
+        for stop in synth_exits {
+            let _ = stop.send(());
+        }
+
+        // Wait for peers to complete
+        for handle in synth_handles {
+            handle.await.unwrap();
+        }
+
+        // Collect stats for this run
+        let mut stats = Stats::new(MAX_PEERS, synth_count);
+        stats.time = test_start.elapsed().as_secs_f64();
+        {
+            let snapshot = test_metrics.take_snapshot();
+
+            stats.accepted = snapshot.get_counter(METRIC_ACCEPTED) as u16;
+            stats.terminated = snapshot.get_counter(METRIC_TERMINATED) as u16;
+            stats.rejected = snapshot.get_counter(METRIC_REJECTED) as u16;
+            stats.conn_error = snapshot.get_counter(METRIC_ERROR) as u16;
+
+            stats.timed_out = synth_count - stats.accepted - stats.rejected - stats.conn_error;
+        }
+        all_stats.push(stats);
+    }
+
+    // TODO: assertions with expected test results.
+
+    node.stop().expect("failed to stop node");

```

Comment:
just for consistency with other files
```suggestion
    node.stop().expect("unable to stop the node");
```